query-grammar/src/query_grammar.rs

```diff
@@ -331,7 +331,8 @@ fn exists_precond(inp: &str) -> IResult<&str, (), ()> {
         peek(tuple((
             field_name,
             multispace0,
-            char('*'), // when we are here, we know it can't be anything but a exists
+            char('*'),
+            peek(alt((multispace1, eof))), // we need to check this isn't a wildcard query
```

Comment:
will that work with something like 

`(toto:*)`

or in a more legitimate case

`(hello AND toto:*) OR happy`

Reply:
it did not, and wouldn't have been possible to test without fixing comment#2